lib/services/local/access_list_test.go

```diff
@@ -1353,6 +1354,51 @@ func TestChangingOwnershipModeIsAnError(t *testing.T) {
 	}
 }
 
+func TestAccessListService_ListAllAccessListMembers(t *testing.T) {
+	ctx := context.Background()
+	clock := clockwork.NewFakeClock()
+
+	mem, err := memory.New(memory.Config{
+		Context: ctx,
+		Clock:   clock,
+	})
+	require.NoError(t, err)
+
+	service := newAccessListService(t, mem, clock, true /* igsEnabled */)
+
+	const numAccessLists = 10
+	const numAccessListMembersPerAccessList = 250
+
+	// Create several access lists.
+	for i := 0; i < numAccessLists; i++ {
+		alName := fmt.Sprintf("%d", i)
+		_, err := service.UpsertAccessList(ctx, newAccessList(t, alName, clock))
+		require.NoError(t, err)
+
+		for j := 0; j < numAccessListMembersPerAccessList; j++ {
+			_, err := service.UpsertAccessListMember(ctx, newAccessListMember(t, alName, fmt.Sprintf("%03d", j)))
+			require.NoError(t, err)
+		}
+	}
+
+	allMembers := make([]*accesslist.AccessListMember, 0, numAccessLists*numAccessListMembersPerAccessList)
+	var nextToken string
+	for {
+		var members []*accesslist.AccessListMember
+		var err error
+		members, nextToken, err = service.ListAllAccessListMembers(ctx, 0, nextToken)
+		require.NoError(t, err)
+
+		allMembers = append(allMembers, members...)
+
+		if nextToken == "" {
+			break
+		}
+	}
+
+	require.Equal(t, numAccessLists*numAccessListMembersPerAccessList, len(allMembers))
```

Comment:
If possible I'd actually check that we got back all and only the members that we expected and in the intended order.